runtime/lua/vim/lsp/util.lua

```diff
@@ -755,7 +755,14 @@ local function create_file(change)
   -- from spec: Overwrite wins over `ignoreIfExists`
   local fname = vim.uri_to_fname(change.uri)
   if not opts.ignoreIfExists or opts.overwrite then
-    local file = io.open(fname, 'w')
+    io.create_missing_dir_open = function(filename, mode)
+        local dir = vim.fs.dirname(filename)
+        if vim.fn.isdirectory(dir) == 0 then
+            vim.fn.mkdir(dir, "p")
+        end
+        return io.open(filename, mode).
+    end
+    local file = io.create_missing_dir_open(fname, 'w')
```

Comment:
```suggestion
    vim.fn.mkdir(vim.fs.dirname(fname), "p")
    local file = io.create_missing_dir_open(fname, 'w')
```

I think this should do the trick too.
No need to create a local function if it's only called once and in the same place.

The `io` module shouldn't be monkey patched.
`mkdir` with `p` exits silently if the directory already exists, so the up-front check isn't necessary either.